lib/utils/datahub.js

```diff
@@ -112,30 +113,91 @@ class DataHub extends EventEmitter {
     throw new Error(responseError(res))
   }
 
-  async pushFlow(specPath){
+  async pushFlow(specPath, dpJsonPath = '.datahub/datapackage.json'){
     let spec = {}
     let descriptor = {}
+
+    if (fs.existsSync(dpJsonPath)) {
+      let content
+      try {
```

Comment:
What for is this try-catch-throw overhead?
```javascript
try {
    content = fs.readFileSync(dpJsonPath)
} catch (err) {
    throw new Error(err.message)
}
```
If the `readFileSync` will got any error, it will be thrown anyway, you are just duplicating the same behavior.

Reply:
Not sure, but that's how errors are handled in entire document. I just followed them

Reply:
It only have sense if you are handling error, or want to change the error message.

Reply:
Also, we probably want to continue even something is wrong with `dp.json`, right?